ToDoTime/ClientApp/src/api/hooks/useGetTasksListDetails.ts

```diff
@@ -1,32 +1,26 @@
-import { BASE_URL, ENDPOINTS_URL } from "api/endpoints";
-import axios from "axios";
-import { useEffect, useState } from "react";
+import { axiosInstance } from "api/axiosInstance";
+import { ENDPOINTS_URL } from "api/endpoints";
+import { ErrorResponse } from "api/types/ErrorResponse";
+import { useQuery } from "react-query";
 import { TasksList } from "shared/types/TasksList";
 
-export const useGetTasksListDetails = (listId: number) => {
-  const [tasksList, setTasksList] = useState<TasksList>();
-  const [isLoading, setIsLoading] = useState(true);
-  const [errorMessage, setErrorMessage] = useState<string>();
-
-  useEffect(() => {
-    if (listId <= 0) return;
+const retrieveListDetail = async (listId: number) => {
+  const response = await axiosInstance.get(ENDPOINTS_URL.getListDetails(listId));
+  return response.data;
+};
 
-    axios
-      .get(`${BASE_URL}${ENDPOINTS_URL.getListDetails(listId)}`)
-      .then((res) => {
-        setTasksList(res.data);
-      })
-      .catch((error) => {
-        setErrorMessage(error.response.data);
-      })
-      .finally(() => {
-        setIsLoading(false);
-      });
-  }, [listId]);
+export const useGetTasksListDetails = (listId: number) => {
+  const {
+    data: tasksList,
+    error,
+    isLoading
+  } = useQuery<TasksList, ErrorResponse>(`tasksListDetailData-${listId}`, () =>
```

Comment:
Should probably make a queryKey helper, otherwise when we start adding tasks we gonna need to write the same key again to either invalidate or add the new data.

Reply:
Fixed